src/System.Linq/tests/AverageTests.cs

```diff
@@ -44,6 +38,25 @@ public void EmptyNullableFloatSource()
         }
 
         [Fact]
+        public void NullNFloatSource()
+        {
+            Assert.Throws<ArgumentNullException>("source", () => default(IEnumerable<float?>).Average());
```

Comment:
@stephentoub it's a bug in the native compiler.  I tried out this minimal repro and verified it warns in the native compiler but not in Roslyn.

``` csharp
    interface ITest { }
    static class Program
    {
        static void M(this ITest t) { }

        static void Main(string[] args)
        {
            default(ITest).M();
        }
    }
```